sqlx-jiff/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+use serde::{Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
+
+#[cfg(feature = "postgres")]
+mod postgres;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Timestamp(pub jiff::Timestamp);
+
+impl From<Timestamp> for jiff::Timestamp {
+    fn from(ts: Timestamp) -> Self {
+        ts.0
+    }
+}
+
+impl From<jiff::Timestamp> for Timestamp {
+    fn from(ts: jiff::Timestamp) -> Self {
+        Self(ts)
+    }
+}
+
+impl Deref for Timestamp {
```

Comment:
`impl AsRef` also?

Reply:
I'm not sure how far we want to go here. In theory, there are a lot of different traits we might want to implement. In practice, I'm not sure how useful they all are or whether we even want to encourage their use. I'd start with a minimal useful set for now.

I'm actually not even sure about using `Deref` here. I'd leave that out for now.

Reply:
OK. Then I'd revoke `#[derive(PartialEq, Eq, Serialize, Deserialize)]` also but just make it an opaque wrapper type. Users should always call `to_jiff` and use `jiff::Timestamp`'s method.

Reply:
@BurntSushi Any further concerns about the wrapping style and integration style? Or shall I continue the integration for other types and adding tests.

Reply:
I honestly don't know. The only way this gets merged is if I have the time to dig into it and figure it whether this is the right approach overall. I don't know when I'm going to have time to do that. Hopefully soon.

Reply:
The problem is that I haven't gone through this exercise yet. So there's no established pattern to follow.

Reply:
Thanks for your feedback. That's enough for the current state.

I need to evaluate whether this patch will be handled at some point so that I can continue to invest my time. As long as there is no certain blocker and we generally accept adding a `jiff-sqlx` crate, that should be fine.

Reply:
I don't have any certain blockers at present.

But just to be clear here, I am going to have run through and do this exercise myself, at least one or two times. I don't know how else I can definitively nail down the requirements. And there will be a lot of documentation to write as well.

I'd be willing to merge something that I know is incomplete and not polished, but I won't release it until I've had a chance to go through it myself.

Reply:
Sure.

> I'd be willing to merge something that I know is incomplete and not polished, but I won't release it until I've had a chance to go through it myself.

This is also my flavor as a maintainer :D